packages/oats-runtime/src/make.ts

```diff
@@ -352,24 +352,56 @@ export function makeOneOf(...options: any[]) {
             return error('multiple preferred options match');
           }
           preferredSuccess = mapped;
-        } else if (success) {
-          return error('multiple options match');
+        } else if (success.length > 0) {
+          if (opts?.unknownField === 'drop') {
+            success.push(mapped);
+          } else {
+            return error('multiple options match');
+          }
         } else {
-          success = mapped;
+          success.push(mapped);
         }
       } else {
         errors = [...errors, mapped];
       }
     }
-    if (preferredSuccess || success) {
-      return preferredSuccess || success;
+    if (preferredSuccess) {
+      return preferredSuccess;
+    }
+
+    if (success.length === 1) {
+      return success[0];
     }
+
+    if (isMultipleMatchesFound(success)) {
```

Comment:
could you motivate why we need this kind of shenanigans?

As is this feels very specific to what you happen to need which would lead me to think that maybe we could achieve the same using makeoptions or even some kind of middleware setup. I kinda would prefer having a time machine and not doing the whole drop thing altogether as it makes thinking about the behaviour way harder for me.

Reply:
Sure thing.

We want to move to a state between s2s communications that all schemas between services are not always 1-to-1. This allows us having looser coupling between services; service A can start to send new property to service B that doesn't still have that definition in its own schema.

As we are using oats for parsing our requests & responses, then using oats types within our service boundaries, this way we can achieve the looser coupling with the guarantee that within our service boundaries objects only have properties defined in the schema itself.

This change would make `drop` behave the same way as `fail` does when matching to schemas which allows services to change the default behaviour of `fail` to `drop` without having to, possibly, change how their schemas are defined.

Reply:
would something like https://github.com/smartlyio/oats/pull/419 do?